src/contractIntegration/interaction/web3Provider/Web3ProviderClam.ts

```diff
@@ -0,0 +1,57 @@
+import Web3 from "web3";
+import { IdentityManager } from "../../../indentityManager";
+import IInteractionConfig from "../IInteractionConfig";
+import IWeb3Provider from "./IWeb3Provider";
+import { injectable, inject } from "tsyringe";
+
+@injectable()
+class Web3ProviderClam implements IWeb3Provider {
+    private provider: Web3;
+    private interactionConfig: IInteractionConfig
+
+    constructor(@inject("Provider") provider: any, @inject("Config") interactionConfig: any) {
+        this.provider = new Web3(provider);
+        this.interactionConfig = interactionConfig;
+    }
+    getMethods(interactionType: string) {
+        let abi: any;
+        let address: string = "";
+        if (interactionType === "consent") {
+            abi = this.interactionConfig.consent.abi;
+            address = this.interactionConfig.consent.address;
+        } else if (interactionType === "access") {
+            abi = this.interactionConfig.access.abi;
+            address = this.interactionConfig.access.address;
+        } else {
+            abi = this.interactionConfig.consent.abi;
```

Comment:
@HectorGerardoZV we should throw an error if the method was not passed

Reply:
Done